eth/beacon/state_machines/base.py

```diff
@@ -193,9 +224,272 @@ def active_state(self) -> ActiveState:
+            chaindb,
+            config,
+        )
+
+        # Set state roots
+        post_block = block.copy(
+            crystallized_state_root=post_crystallized_state.hash,
+            active_state_root=post_active_state.hash,
+        )
+        filled_block_proposal = BlockProposal(
+            block=post_block,
+            shard_id=block_proposal.shard_id,
+            shard_block_hash=block_proposal.shard_block_hash,
+        )
+
+        proposer_attestation = self.attest_proposed_block(
+            post_crystallized_state,
+            post_active_state,
+            active_state,
+            filled_block_proposal,
+            chaindb,
+            config.CYCLE_LENGTH,
+            private_key,
+        )
+        return post_block, post_crystallized_state, post_active_state, proposer_attestation
+
+    def _update_the_states(self,
+                           crystallized_state: CrystallizedState,
+                           active_state: ActiveState) -> None:
+        self._crytallized_state = crystallized_state
+        self._active_state = active_state
+
+    def attest_proposed_block(self,
+                              post_crystallized_state: CrystallizedState,
+                              post_active_state: ActiveState,
+                              pre_active_state: ActiveState,
+                              block_proposal: 'BlockProposal',
+                              chaindb: BaseBeaconChainDB,
+                              cycle_length: int,
+                              private_key: int) -> 'AttestationRecord':
+        """
+        Return the initial attestation by the block proposer.
+
+        The proposer broadcasts their attestation with the proposed block.
+        """
+        block_committees_info = get_block_committees_info(
+            block_proposal.block,
+            post_crystallized_state,
+            cycle_length,
+        )
+        # Vote
+        attester_bitfield = set_voted(
+            get_empty_bitfield(block_committees_info.proposer_committee_size),
+            block_committees_info.proposer_index_in_committee,
+        )
+
+        # Get justified_slot and justified_block_hash
+        justified_slot = post_crystallized_state.last_justified_slot
+        justified_block_hash = chaindb.get_canonical_block_hash_by_slot(justified_slot)
+
+        # Get signing message and sign it
+        parent_hashes = get_hashes_to_sign(
+            post_active_state.recent_block_hashes,
+            block_proposal.block,
+            cycle_length,
+        )
+
+        message = create_signing_message(
+            block_proposal.block.slot_number,
+            parent_hashes,
+            block_proposal.shard_id,
+            block_proposal.shard_block_hash,
+            justified_slot,
+        )
+        sigs = [
+            bls.sign(
+                message,
+                private_key,
+            )
+        ]
+        aggregate_sig = bls.aggregate_sigs(sigs)
```

Comment:
No need to aggregate the single sig. A sig and an aggregate sig are indistinguishable in BLS